src/influx_udp_worker.erl

```diff
@@ -77,66 +88,124 @@ code_change(_OldVsn, State, _Extra) ->
 
 send_data(_, false) -> {error, wrong_data};
 
-send_data(#state{socket=Socket, port=Port, host=Host}, Data) ->
+send_data(#state{socket=Socket, port=Port, host=Host, debug = Debug}, Data) ->
+  debug_send(Data, Debug),
   gen_udp:send(Socket, Host, Port, Data).
 
 
 %% @doc
-%% Convert data to JSON (binary) of a type:
-%% [
-%%    {
-%%        "name": "my_series",
-%%        "columns": ["ip", "name"],
-%%        "points": [
-%%            ["127.0.0.1", "John"],
-%%            ["192.168.0.1", "Paul"]
-%%        ]
-%%    }
-%% ]
+%% Convert data to line protocol:
+%% my_series,host=eu-west,ip="127.0.0.1" value=43 10000000000
+%% my_series,host=eu-west,ip="192.168.0.1" value=54 10000000000
 %% @end
 -spec encode(
   Name::string()|atom()|binary(),
-  Points::list(influx_data_point())|influx_data_point()
-) -> JSON::binary().
-encode(Name, Points) when is_list(Name) ->
-  encode(to_key(Name), Points);
-
-encode(Name, Points) ->
+  Points::list(influx_data_point())|influx_data_point(),
+  Tags::proplists:proplist()|map()
+) -> Line::binary().
+encode(Name, Points, Tags) ->
+  BTags = remove_trailing_comma(
+    extract_tags(Tags)
+  ),
+  BName = to_key(Name),
+  Prefix = <<BName/binary, BTags/binary>>,
   case encode_points(Points) of
-    {[], _} -> false;
-    {Columns, Values} -> 
-      jsx:encode(
-        [#{
-            name => Name,
-            columns => normalize_columns(Columns),
-            points => Values
-          }
-        ]
+    [] -> false;
+    Values -> 
+      lists:foldl(
+        fun(V, Acc) ->
+          <<Acc/binary, Prefix/binary, ?SPACE/binary, V/binary, ?NEW_LINE/binary>>
```

Comment:
Line 117 is too long:           <<Acc/binary, Prefix/binary, ?SPACE/binary, V/binary, ?NEW_LINE/binary>>.